src/iam.ts

```diff
@@ -215,11 +215,29 @@ export class IAM extends IAMBase {
     includeClaims = true
   }: { did?: string; includeClaims?: boolean } | undefined = {}) {
     if (this._cacheClient && did) {
-      const didDoc = await this._cacheClient.getDidDocument({ did, includeClaims });
-      return {
-        ...didDoc,
-        service: didDoc.service as (IServiceEndpoint & ClaimData)[]
-      };
+      try {
+        const didDoc = await this._cacheClient.getDidDocument({ did, includeClaims });
+        return {
+          ...didDoc,
+          service: didDoc.service as (IServiceEndpoint & ClaimData)[]
+        };
+      } catch (err) {
+        if (err?.response?.status === 404) {
```

Comment:
I don't think we could catch only on 404, I think we should catch any error and just fallback to the blockchain. Does that seem reasonable?

Reply:
Ok, let's fallback to blockchain on every error except 401

Reply:
I think there's a case to be made for getting from chain even if you get a 401 from the cache-server. To me, it doesn't really sense that iam-client-lib would say "oh, you're not authorized for the cache-server, so I'm not going to let you fallback to the blockchain directly". What do you think?

Reply:
Yeah, your right 👍  there is no need to be authorized to get something from chain, I'll change that